candle-core/src/indexer.rs

```diff
@@ -46,6 +46,40 @@ impl Tensor {
                     current_dim += 1;
                     out
                 }
+                TensorIndexer::Step(left_bound, right_bound, step) => {
+                    let start = match left_bound {
+                        Bound::Included(n) => *n,
+                        Bound::Excluded(n) => *n + 1,
+                        Bound::Unbounded => 0,
+                    } as i64;
+                    let stop = match right_bound {
+                        Bound::Included(n) => *n + 1,
+                        Bound::Excluded(n) => *n,
+                        Bound::Unbounded => dims[i],
+                    } as i64;
+
+                    let step = *step as i64;
+                    let device = x.device();
+                    let indices = match step {
+                        1.. => Tensor::arange_step(start as i64, stop as i64, step, device)?,
+                        ..=-1 => {
+                            let mut data = vec![];
+                            let mut current = stop - 1;
+                            while current >= start {
+                                data.push(current);
+                                current += step;
+                            }
+                            let len = data.len();
+
+                            Tensor::from_vec_impl(data, len, device, false)?
+                        }
+                        0 => panic!("step size cannot be zero"),
```

Comment:
Let's use `candle::bail!` rather than `panic!` here so that it's easier to catch the error.